app/experimenter/docs/adrs/0002-use-react-bootstrap-cra-gql-for-nimbus.md

```diff
@@ -0,0 +1,113 @@
+# Use React, Bootstrap, CRA, and GQL for the Nimbus Front-End Stack
+
+* Status: proposed
+* Deciders: Lauren Zugai, Jody Heavener, [TBD]
+* Date: 2020-09-08
+
+## Context and Problem Statement
+
+The front-end for Experimenter is currently composed of two applications. A more mature application for Normandy experiments, found at `app/experimenter/static/core`, is composed of a mix of Django templates, vanilla JS, jQuery, and React components. A newer React application for Nimbus experiments (previously named Rapid experiments) can be found at `app/experimenter/static/rapid`.
+
+With the decision made to build the Nimbus console as part of the Experimenter codebase ([see ADR #0001](https://github.com/mozilla/experimenter/blob/main/app/experimenter/docs/adrs/0001-build-nimbus-console-in-experimenter.md)), it was determined we likely want a "new front-end." This ADR, while slightly atypical as it does not serve to document a single architectural change, outlines a proposed directory tree change regarding the front-end, a proposed new front-end stack, how it compares to the current `rapid` stack, and how we will benefit from these changes.
+
+## Decision Drivers
+
+Drivers for directory tree changes:
+* A more obvious separation of concerns for:
+  * the front-end and back-end
+  * legacy front-end code and Nimbus front-end code
+
+Drivers for the proposed new stack:
+* Better development tooling
+* A more cohesive fit into the FxA team's stack and processes
+* Development speed
+
+Drivers for starting a new application:
+* Time needed and ease of refactoring the current `rapid` React application vs time needed to start with a new application
+
+## Proposed Changes
+
+### App Directory Tree
+
+Currently, a `static/` directory contains the front-end applications with some shared assets, and a `templates/` directory lives adjacent to it.  To provide a more obvious separation of concerns in the directory tree:
```

Comment:
Yeah, TBH I like both of those suggestions (probably the latter the best), though I hesitate due to needing to serve with nginx just for project structure. Regarding "client," I actually called it "client" at first, but I think @k88hudson said "client" sounds like the SDK. 🤔

I think `app/client/` and `app/server/` is pretty slick, but are there any long-term downsides to using nginx to serve the application besides an initial setup and is "this is slightly better for separation of concern" a good enough reason to justify it (and the sweeping filename changes from `experimenter` to `server`)?

Reply:
Well for the server I think we'd just move the entire `app/experimenter/` dir to `app/server/experimenter/` whole sale so we woudln't need to rename anything.  We'd only have to rename things if we did it the other way around `app/experimenter/server/` if we wanted to put client inside the Django top level.  If we put the client outside the Django top level to serve it with nginx we don't need to rename anything.

There's nothing stopping us from putting the entire frontend outside of Django and only letting nginx know about it, but we'd have to change a couple steps for local dev and prod

1) Expose the built assets (`dist`) in a shared volume to the nginx container
2) Add a rule to nginx to host that on some path
3) Hard code that path in the templates to initialize it like 
```html
  <script src="/<nginx-nimbus-asset-path>/index.js"></script>
```

 rather than using the Django static asset path resolver `{% static "assets/rapid/main.js" %}` see [here](https://github.com/mozilla/experimenter/blob/main/app/experimenter/templates/experiments/rapid.html#L10)

Reply:
@jaredlockhart I'm thinking at present, it may not be worth serving the FE with nginx for this degree of FE/BE separation due to the additional work needed. I personally am OK with `app/experimenter/nimbus-ui`, as in my mind it solves my initial "I want more separation between FE/BE + nimbus/legacy" problem. We could also potentially do this later if we wanted to. I'm writing up a section in this ADR about considering this, thank you for the breakdown. Feel free to disagree (@jodyheavener  might argue it's worth it). 🙂

Reply:
That would be the simplest approach to start 👍

Reply:
Yeah, I generally think it would be good to maintain a clear separation, and I'm sure there are merits to doing it out of the gate versus refactoring later, but I agree that this might be more effort than it's currently worth if we're trying to reach MVP sooner rather than later. I'm cool with `app/experimenter/nimbus-ui`.